.github/workflows/build-nightly.yaml

```diff
@@ -210,3 +210,112 @@ jobs:
           DATACORDER_USER: ${{ secrets.DATACORDER_USER }}
           DATACORDER_SSHPASS: ${{ secrets.DATACORDER_PASSWORD }}
         run: $GITHUB_WORKSPACE/ci/linux/upload_dist_package.sh nightly
+
+  build_mac:
+    strategy:
+      matrix:
+        configuration: [FastDebug, Release]
+        arch: [x86_64]
+    name: Mac
+    runs-on: macos-latest
+    steps:
+      - name: Cache Qt
+        id: cache-qt-lin
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}/../Qt
+          key: ${{ runner.os }}-QtCache-${{ env.QT_VERSION }}
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          version: ${{ env.QT_VERSION }}
+          dir: ${{ github.workspace }}/..
+          cached: ${{ steps.cache-qt-lin.outputs.cache-hit }}
+          setup-python: 'false'
+          aqtversion: ==1.1.3
+
+      - uses: actions/checkout@v1
+        name: Checkout
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Set up test version
+        shell: bash
+        run: |
+          [[ "${{ github.ref }}" =~ ^refs\/heads\/test\/(.*)$ ]]
+          # Override the revision string so that the builds are named correctly
+          echo "set(FSO_VERSION_REVISION_STR ${BASH_REMATCH[1]})" > "version_override.cmake"
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ runner.os }}-${{ matrix.configuration }}-${{ matrix.compiler }}
+
+      - name: Configure CMake
+        env:
+          CONFIGURATION: ${{ matrix.configuration }}
+          COMPILER: ${{ matrix.compiler }}
+          ARCHITECTURE: ${{ matrix.arch }}
+          JOB_CMAKE_OPTIONS: ${{ matrix.cmake_options }}
+        run: |
```

Comment:
`mfpmath` is not recognized on Mac Clang. Platform is specified via `CMAKE_OSX_ARCHITECTURES`.

Reply:
Should I copy `linux/configure_cmake.sh` to `mac/configure_cmake.sh` and make changes? Or reuse the Linux scripts and make changes to those?

Reply:
I see that Windows is not using scripts and is passing configuration in the CI config, so I wasn't sure what is the correct way for Mac.

Reply:
Ok, there's an organization decision to be made here. `ci/linux` has scripts that run on linux runners, but are not exclusively for linux build jobs.  There's scripts, for example, that run on the linux runner that do the publishing.

We can either make a duplicate of `linux/configure_cmake.sh` and make the mac specific changes to `mac/configure_cmake.sh` or modify `linux/configure_cmake.sh` to handle both linux and mac builds.

I'd prefer the former, but if that causes problems for you then we can go with `mac/configure_cmake.sh`

And yeah, the windows/linux commonality between the build jobs is a tad messy.  I need to get around to making a common build workflow for all the variations to cut down on the inconsistencies.

Reply:
Do you know if there is an env variable that is available by default that would allow the script to figure out if it's running on a Mac or a Linux runner?

Reply:
[See here](https://docs.github.com/en/actions/learn-github-actions/environment-variables#detecting-the-operating-system)

Reply:
Updated.